source/ControllableHero.hx

```diff
@@ -0,0 +1,54 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+
+using flixel.util.FlxSpriteUtil;
+
+// Part 2: Added a ControllableHero Class
+class ControllableHero extends FlxSprite
+{
+	// Movement variable
+	var speed = 10; // Couldn't figure out a better way to handle this?
+
+	// Implement Constructor
+	public function new()
+	{
+		// Invoke parent class constructor
+		super(x, y);
+		this.x = x;
+		this.y = y;
+
+		// Make a simple rectangle sprite
+		var myRectangle = new FlxSprite(x, y);
+		makeGraphic(30, 10, 0xFFFFA500, true); // FlxColor Orange
+	}
+
+	// Override inherited update() function/method
+	override public function update(elapsed:Float)
+	{
+		keyMovement();
+		super.update(elapsed);
+	}
+
+	// Keyboard Movement function/method
+	public function keyMovement()
+	{
+		if (FlxG.keys.anyPressed([UP, W]))
+		{
+			this.y -= Math.round(speed);
```

Comment:
this is fine, though it still has the problem of tying our game logic to the framerate (which isn't a problem for this assignment, but something we want to improve in the future).

If you want to use the built-in velocity handling from `FlxObject`, you could instead do something like
```
velocity.set(0,0);
if (FlxG.keys.anyPressed([UP, W]))
{
    velocity.y -= speed;
}
...
```
Every time update is called, velocity is set back to 0, then x and y velocity is changed according to which inputs are pressed. This does mean that a player moving diagonally covers more total distance (10 vs 10 * sqrt(2)) than just moving in a cardinal direction, but it does effectively separate our game logic from the visual framerate.